src/ring_theory/ideal/local_ring.lean

```diff
@@ -106,24 +74,56 @@ begin
   rwa ←eq_maximal_ideal hM1
 end
 
-@[simp] lemma mem_maximal_ideal (x) :
-  x ∈ maximal_ideal R ↔ x ∈ nonunits R := iff.rfl
+@[simp] lemma mem_maximal_ideal (x) : x ∈ maximal_ideal R ↔ x ∈ nonunits R := iff.rfl
+
+end comm_semiring
+
+section comm_ring
+variables {R : Type u} [comm_ring R] [local_ring R]
```

Comment:
You can move this line at the beginning of the file, so you only write `{R : Type u} {S : Type v}` once, and then you only need `[comm_ring]` and similar.